### PR TITLE
Visit all network class not just direct sub classes.

### PR DESCRIPTION
**Pull request by Yannig**
##### ISSUE TYPE
- Bugfix Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel_fix_network_facts_solaris 4e7ad813e7) last updated 2016/04/16 19:19:33 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 5409ed1b28) last updated 2016/04/16 19:14:11 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 3afe117730) last updated 2016/04/16 19:14:11 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Visit all network class not just direct sub classes while gathering facts.

```
before:
solaris | SUCCESS => {
    "ansible_facts": {}

after:
solaris | SUCCESS => {
    "ansible_facts": {
        "ansible_net0": {
            "device": "net0", 
            "ipv4": [
                {
                    "address": "10.0.2.15", 
                    "broadcast": "10.0.2.255", 
                    "flags": [
...
                    ], 
                    "mtu": "1500", 
                    "netmask": "255.255.255.0", 
                    "network": "10.0.2.0"
                }
            ], 
...
```

Fix for https://github.com/ansible/ansible/issues/15446

### CODE REVIEW

**Comment by abadger** [comment]
Looks good -- we may need to do this same thing in basic.py as well: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L306

Do you feel like pulling this code out into a function in module_utils/basic.py that can be used by both facts and basic?

**Comment by Yannig** [comment]
@abadger no problem :). I'll try to do this week.

File: lib/ansible/module_utils/facts.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import getpass
 import pwd
 import ConfigParser
+from basic import get_all_subclasses
```
Comment (abadger) [inline lib/ansible/module_utils/facts.py:35]: Need to change this to an absolute import due to the way ziploader works

``` python
from ansible.module_utils.basic import get_all_subclasses
```

**Comment by abadger** [comment]
Looks good.  Just one minor thing to change.  I'll make that change and merge.

**Comment by abadger** [comment]
Merged in this sequence of commits:
- eb18767f91c2409737bbeb345880379c6124977e
- 72f17f3ff3c9fb8451c981cfe71493b3fa3ede29
- d7189a442efad46947f02b3cd1306efba60df06d

Thanks @Yannig !

**Pull request closed without merging**